component/src/main/java/org/wso2/carbon/identity/authenticator/FoursquareAuthenticator.java

```diff
@@ -30,322 +30,367 @@
+			throws AuthenticationFailedException {
+		try {
+			Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+			String clientId = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID);
+			String clientSecret = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_SECRET);
+			String tokenEndPoint = getTokenEndpoint(authenticatorProperties);
+			if (tokenEndPoint == null) {
+				tokenEndPoint = authenticatorProperties.get(FoursquareAuthenticatorConstants.OAUTH2_TOKEN_URL);
+			}
+			String callbackurl = getCallbackUrl(authenticatorProperties);
+			OAuthAuthzResponse authzResponse = OAuthAuthzResponse.oauthCodeAuthzResponse(request);
+			String code = authzResponse.getCode();
+			OAuthClientRequest accessRequest;
+			try {
+				accessRequest =
+						OAuthClientRequest.tokenLocation(tokenEndPoint).setGrantType(GrantType.AUTHORIZATION_CODE)
+						                  .setClientId(clientId).setClientSecret(clientSecret)
+						                  .setRedirectURI(callbackurl).setCode(code).buildBodyMessage();
+				// create OAuth client that uses custom http client under the hood
+				OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
+				OAuthClientResponse oAuthResponse;
+				oAuthResponse = oAuthClient.accessToken(accessRequest);
+				String accessToken = oAuthResponse.getParam(FoursquareAuthenticatorConstants.ACCESS_TOKEN);
+				if (StringUtils.isNotEmpty(accessToken)) {
+					Map<ClaimMapping, String> claims = buildClaims(oAuthResponse, authenticatorProperties);
+					if (claims != null && !claims.isEmpty()) {
+						//Find the subject from the IDP claim mapping, subject Claim URI.
+						String subjectFromClaims = FrameworkUtils
+								.getFederatedSubjectFromClaims(context.getExternalIdP().getIdentityProvider(), claims);
+						associateSubjectFromClaims(context, subjectFromClaims, claims);
+					} else {
+						throw new AuthenticationFailedException(
+								"Claims for the user not found for access Token : " + accessToken);
+					}
+				} else {
+					throw new AuthenticationFailedException("Could not receive a valid access token from Foursquare");
+				}
+			} catch (OAuthSystemException e) {
+				throw new AuthenticationFailedException("Exception while building access token request", e);
+			} catch (ApplicationAuthenticatorException e) {
+				throw new AuthenticationFailedException("Exception while building the claim mapping", e);
+			}
+		} catch (OAuthProblemException e) {
+			throw new AuthenticationFailedException("Exception while getting the access token form the response", e);
+		}
+	}
+
+	/**
+	 * This method is to get the Foursquare user details.
+	 *
+	 * @param url         user info endpoint.
+	 * @param accessToken access token.
+	 * @return user info
+	 * @throws ApplicationAuthenticatorException
+	 */
+	private JSONObject fetchUserInfo(String url, String accessToken) throws ApplicationAuthenticatorException {
```

Comment:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-foursquare/pull/2/commits/58520246ef2e797650ce96df5d6f5fbde73cfa2d